runner.py

```diff
@@ -787,10 +809,38 @@ def setup_services(self):
             if 'cmd' in service:  # must come last
                 docker_run_string.append(service['cmd'])
 
+            # before starting the container, check if the dependent containers are "ready"
+            # if not, wait for them
+            if 'depends_on' in service:
+                for dependent_container in service['depends_on']:
```

Comment:
It would be nice to have a condition here that checks if the long form or the short form is used.

The short form is how this PR currently behaves. However the long form would also have features like conditions, health_checks etc.

I imagine a condition check that then throws a RuntimeError if the long form is used.

However this can be "overridden" by supplying an argument to the runner like `--skip-unsupported`.

Just not using the variables silently might introduce quite some confusion. What do you think?

Actually we could use this flag then in an extended form and throw this warning for every key in the `compose.yml` that we do not support 🤔

Reply:
Created a new issue for this:
https://github.com/green-coding-berlin/green-metrics-tool/issues/580